perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/ExpirationNotifSchedulerTest.java

```diff
@@ -232,6 +239,70 @@ public void checkMembersState() throws Exception {
 
 	}
 
+	@Test
+	public void checkVoApplicationShouldBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkVoApplicationShouldBeAutoRejected");
+
+		Application submitApp = setUpAndSubmitAppForPotentialAutoRejection(70, null, VO_APP_EXP_RULES);
+
+		ReflectionTestUtils.invokeMethod(spyScheduler, "voApplicationsAutoRejection", perun.getVosManagerBl().getVos(session));
+
+		// check results
+		Application returnedApp = registrarManager.getApplicationById(session, submitApp.getId());
+		assertEquals("Application should be rejected.", returnedApp.getState(), Application.AppState.REJECTED);
+	}
+
+	@Test
+	public void checkVoApplicationShouldNotBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkVoApplicationShouldNotBeAutoRejected");
+
+		Application submitApp = setUpAndSubmitAppForPotentialAutoRejection(50, null, VO_APP_EXP_RULES);
+
+		ReflectionTestUtils.invokeMethod(spyScheduler, "voApplicationsAutoRejection", perun.getVosManagerBl().getVos(session));
+
+		// check results
+		Application returnedApp = registrarManager.getApplicationById(session, submitApp.getId());
+		assertEquals("Application shouldn't be rejected.", returnedApp.getState(), Application.AppState.VERIFIED);
+	}
+
+	@Test
+	public void checkGroupApplicationShouldBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkGroupApplicationAutoRejection");
+
+		jdbc.update("DELETE from application_form WHERE vo_id = ?", vo.getId());
```

Comment:
I don't get that. This deletes all vo and group applications of that VO. Shouldn't each test have own rollbacked transaction? Why do you do this?

Reply:
This is actually necessary because the HSQLDB has a slightly different schema. In the regular postgres.sql we have: 
```
create unique index applform_u1 on application_form (vo_id) where group_id is null;
create unique index applform_u2 on application_form (vo_id, group_id) where group_id is not null;
```
But in the test-schema.sql:
```
create unique index applform_u1 on application_form (vo_id);
create unique index applform_u2 on application_form (vo_id, group_id);
```
It seems that HSQLDB doesn't support the `where` condition. Because of that, in tests, there can be only a single application form across the whole Vo and its groups.

So, to create an application form in a group, you have to delete the form for its vo.

Reply:
I couldn't figure out a better solution, so I suggested this workaround for now.

Reply:
Wow, didn't know that. Maybe this note should be in the test method and it should be called only for HSQLDB, since we sometime run tests against real DB, where this condition doesn't apply.